### PR TITLE
Deprecate DSA harder.

### PR DESCRIPTION
**Pull request by paulidale**
This deprecates all functions that deal with the types DSA and DSA_SIG.

- [x] documentation is added or updated
- [x] tests are added or updated

### CODE REVIEW

File: include/openssl/pem.h [hunk 1]
```diff
@@ -455,7 +457,7 @@ EVP_PKEY *PEM_read_bio_Parameters_ex(BIO *bp, EVP_PKEY **x,
 EVP_PKEY *PEM_read_bio_Parameters(BIO *bp, EVP_PKEY **x);
 int PEM_write_bio_Parameters(BIO *bp, const EVP_PKEY *x);
 
-# ifndef OPENSSL_NO_DSA
+# if !defined(OPENSSL_NO_DSA) || !defined(OPENSSL_NO_DEPRECATED_3_0)
```
Comment (levitte) [inline include/openssl/pem.h:460]: Break this into two condition lines.  OpenSSL::ParseC doesn't have such advanced expression processor that it handles this properly, so `util/mknum.pl` may fail to detect these feature indicators.

Comment (levitte) [reply include/openssl/pem.h:460]: Also, if you're deprecating these functions, please add the deprecation attribute.

**Comment by levitte** [comment]
Why does this depend on #13096?

**Comment by paulidale** [comment]
`OSSL_ENCODER_CTX_new_by_EVP_PKEY()`

**Comment by levitte** [comment]
> There are some other documentation changes that happen in #13096 which this depends on `d2i_RSAPrivateKey.pod` e.g.

Ah!  Yeahok, makes sense

File: include/openssl/pem.h (hunk 1 above)
Comment (paulidale) [reply include/openssl/pem.h]: Done in a few places.

**Comment by kroeckx** [comment]
This fixes #13121?

**Comment by levitte** [comment]
#13096 is merged, so this should be rebased

**Comment by levitte** [comment]
Please rebase this

**Comment by levitte** [comment]
Upon request, I've taken it upon me to finish this PR.  It will appear as another PR, at which point this one will be closed.

**Comment by levitte** [comment]
Closed in favor of #13638

**Pull request closed without merging**